Tickets/dm 30714

pcortes80 opened the pull request.

teweitsai reviewed (approved):
This PR is good to me. However, I do have something want to discuss.

[testSendErrorExplanation.vi]

1. I think we should try to test the detail instead of the type only. I am thinking to test the received message contains a specific error code or not.

<<TcpServerBase.lvclass>>

[sendErrorExplanation.vi]

1. I do not think the function description should have the idea of "Component". Please consider to rephrase it. In addition, you should say the error will be cleared in the description.

2. Please considet to rename the vi to be "clearErrorAndSendExplanation.vi".

3. I think because the error will be cleared, you may need to think which vi should call it to simplify the code. In this case, only the vi in state machine in command and telemetry servers should call this function. This will make the code easier to maintain. Let us discuss this if you do not understand my meaning.

[msgDetailsToJSON.vi]

1. Please consider to remove the TcpServerBase.sendErrorExplanation.vi. Let use discuss the detail of this.

[tcpServerSenderLoop.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.


<<TcpServerCmd.lvclass>>

[connStatusCluster.ctl]

1. Why we have two copies of this ctl?

[eventToClient.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

[getSeverNotifier.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

[processPacket.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.


<<TcpServerTel.lvclass>>

[enqueueTelToClient.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

[enqueueTelToComponent.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

[getSeverNotifier.vi]

1. Why this is different from TcpServerCmd.getSeverNotifier.vi for the error message?

[processTel.vi]

1. We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

2. There is a weird labve of "A".


<<TcpServer.lvclass>>

[configServer.vi]

1. In the label of block diagram, the first word should begin from Capital alphabet.

pcortes80 commented on src/README.md:
```diff
@@ -113,3 +113,5 @@ To set this as **un-block call** use telQueueServerTimeout greater than 0 and En
 - Every time Python TCP Client (cmd and tel) and TCP Server connections are "alive" TCP Server will send an user event to the LabVIEW Component.
 If either command or telemetry instance of Python TCP Client disconnects from TCP Server, a generic user event is sent.
 If all Python TCP Client instances are connected again, a generic user event is sent.
+- The `TcpServerBase.sendErrorExplanation.vi` is used to send an specific and important error to the ComponentSimulator.
```
Fixed. Thanks!

pcortes80 commented:
This PR is good to me. However, I do have something want to discuss.

[testSendErrorExplanation.vi]

I think we should try to test the detail instead of the type only. I am thinking to test the received message contains a specific error code or not.

**Reply:**
- Code fixed as discussed in last meeting.
- In adition, the name was changed to testClearErrorAndSendExplanation.vi.

<<TcpServerBase.lvclass>>

[sendErrorExplanation.vi]

I do not think the function description should have the idea of "Component". Please consider to rephrase it. In addition, you should say the error will be cleared in the description.

**Reply:**
- Description updated as recommended.

Please consider to rename the vi to be "clearErrorAndSendExplanation.vi".

**Reply:**
- Name changed as recommended.

I think because the error will be cleared, you may need to think which vi should call it to simplify the code. In this case, only the vi in state machine in command and telemetry servers should call this function. This will make the code easier to maintain. Let us discuss this if you do not understand my meaning.

**Reply:**
- This is solved in several VIs.

[msgDetailsToJSON.vi]

Please consider to remove the TcpServerBase.sendErrorExplanation.vi. Let use discuss the detail of this.

**Reply:**
- Removed this vi and put in in the recommended place, as we discussed in the meeting. The main idea is put this vi inside of processes, not in a function.

[tcpServerSenderLoop.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- Put it in a recommended place.

<<TcpServerCmd.lvclass>>

[connStatusCluster.ctl]

Why we have two copies of this ctl?

**Reply:**
- I certanly made a mistake with this dependences. It is fixed already.

[eventToClient.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- The `clearErrorAndSendExplanation.vi` was removed and put in the caller vi (TcpServerCmd.runServer.vi).

[getSeverNotifier.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- The `clearErrorAndSendExplanation.vi` was removed and put in the caller vi.

[processPacket.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- The `clearErrorAndSendExplanation.vi` was removed and put in the caller vi (TcpServerCmd.tcpServerCallBack.vi).


<<TcpServerTel.lvclass>>

[enqueueTelToClient.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- this vi is called by the user, so the user will be responsible for deal with this error. I will add a comment in the caller vi description to the user be careful with this.

[enqueueTelToComponent.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- this vi is called by the user, so the user will be responsible for deal with this error. I will add a comment in the caller vi description to the user be careful with this.

[getSeverNotifier.vi]

Why this is different from TcpServerCmd.getSeverNotifier.vi for the error message?

**Reply:**
- I created two instances of this vi in TcpServerCmd and TcpServertel classes to don´t expose them to the User in TcpServer class. I added a comment int the description.

[processTel.vi]

We may need to consider where to put the TcpServerBase.sendErrorExplanation.vi to make the code robust. Let use discuss the detail of this.

**Reply:**
- this vi is called by the user, so the user will be responsible for deal with this error. I will add a comment in the caller vi description to the user be careful with this.

There is a weird labve of "A".

**Reply:**
- Removed.

<<TcpServer.lvclass>>

[configServer.vi]

In the label of block diagram, the first word should begin from Capital alphabet.

**Reply:**
- Fixed.
